### ISSUE TITLE
Deleting all buckets rather than hardcoded buckets

### ISSUE THREAD

**Issue by tleyden** [couchbaselabs/mobile-testkit#390]
When running the sg-replicate tests, I'm seeing this error:

```

TASK [debug] *******************************************************************
ok: [cb1] => {
    "msg": "Couchbase bucket RAM 1440"
}

TASK [Delete buckets] **********************************************************
failed: [cb1] => (item=data-bucket) => {"changed": true, "cmd": "/opt/couchbase/bin/couchbase-cli bucket-delete --cluster=ec2-54-87-254-3.compute-1.amazonaws.com:8091 --user=Administrator --password=password --bucket=data-bucket", "delta": "0:00:00.226985", "end": "2016-04-14 19:31:53.051517", "failed": true, "item": "data-bucket", "rc": 2, "start": "2016-04-14 19:31:52.824532", "stderr": "", "stdout": "ERROR: unable to bucket-delete; (404) Object Not Found\nRequested resource not found.", "stdout_lines": ["ERROR: unable to bucket-delete; (404) Object Not Found", "Requested resource not found."], "warnings": []}
failed: [cb1] => (item=index-bucket) => {"changed": true, "cmd": "/opt/couchbase/bin/couchbase-cli bucket-delete --cluster=ec2-54-87-254-3.compute-1.amazonaws.com:8091 --user=Administrator --password=password --bucket=index-bucket", "delta": "0:00:00.220398", "end": "2016-04-14 19:31:55.346501", "failed": true, "item": "index-bucket", "rc": 2, "start": "2016-04-14 19:31:55.126103", "stderr": "", "stdout": "ERROR: unable to bucket-delete; (404) Object Not Found\nRequested resource not found.", "stdout_lines": ["ERROR: unable to bucket-delete; (404) Object Not Found", "Requested resource not found."], "warnings": []}
...ignoring

TASK [Create new buckets] ******************************************************
failed: [cb1] => (item=data-bucket) => {"changed": true, "cmd": "/opt/couchbase/bin/couchbase-cli bucket-create --cluster=ec2-54-87-254-3.compute-1.amazonaws.com:8091 --user=Administrator --password=password --bucket=data-bucket --bucket-type=couchbase --bucket-port=11211 --bucket-ramsize=1440 --bucket-replica=1 --wait", "delta": "0:00:00.221044", "end": "2016-04-14 19:31:58.630965", "failed": true, "item": "data-bucket", "rc": 2, "start": "2016-04-14 19:31:58.409921", "stderr": "", "stdout": "ERROR: unable to bucket-create; (400) Bad Request\n{\"errors\":{\"ramQuotaMB\":\"RAM quota specified is too large to be provisioned into this cluster.\",\"replicaNumber\":\"Warning: you do not have enough data servers to support this number of replicas.\"},\"summaries\":{\"ramSummary\":{\"total\":3020947456,\"otherBuckets\":3019898880,\"nodesCount\":1,\"perNodeMegs\":1440,\"thisAlloc\":1509949440,\"thisUsed\":0,\"free\":-1508900864},\"hddSummary\":{\"total\":64408858624,\"otherData\":2563062770,\"otherBuckets\":13291574,\"thisUsed\":0,\"free\":61832504280}}}", "stdout_lines": ["ERROR: unable to bucket-create; (400) Bad Request", "{\"errors\":{\"ramQuotaMB\":\"RAM quota specified is too large to be provisioned into this cluster.\",\"replicaNumber\":\"Warning: you do not have enough data servers to support this number of replicas.\"},\"summaries\":{\"ramSummary\":{\"total\":3020947456,\"otherBuckets\":3019898880,\"nodesCount\":1,\"perNodeMegs\":1440,\"thisAlloc\":1509949440,\"thisUsed\":0,\"free\":-1508900864},\"hddSummary\":{\"total\":64408858624,\"otherData\":2563062770,\"otherBuckets\":13291574,\"thisUsed\":0,\"free\":61832504280}}}"], "warnings": []}
failed: [cb1] => (item=index-bucket) => {"changed": true, "cmd": "/opt/couchbase/bin/couchbase-cli bucket-create --cluster=ec2-54-87-254-3.compute-1.amazonaws.com:8091 --user=Administrator --password=password --bucket=index-bucket --bucket-type=couchbase --bucket-port=11211 --bucket-ramsize=1440 --bucket-replica=1 --wait", "delta": "0:00:00.236746", "end": "2016-04-14 19:32:00.286670", "failed": true, "item": "index-bucket", "rc": 2, "start": "2016-04-14 19:32:00.049924", "stderr": "", "stdout": "ERROR: unable to bucket-create; (400) Bad Request\n{\"errors\":{\"ramQuotaMB\":\"RAM quota specified is too large to be provisioned into this cluster.\",\"replicaNumber\":\"Warning: you do not have enough data servers to support this number of replicas.\"},\"summaries\":{\"ramSummary\":{\"total\":3020947456,\"otherBuckets\":3019898880,\"nodesCount\":1,\"perNodeMegs\":1440,\"thisAlloc\":1509949440,\"thisUsed\":0,\"free\":-1508900864},\"hddSummary\":{\"total\":64408858624,\"otherData\":2563062770,\"otherBuckets\":13291574,\"thisUsed\":0,\"free\":61832504280}}}", "stdout_lines": ["ERROR: unable to bucket-create; (400) Bad Request", "{\"errors\":{\"ramQuotaMB\":\"RAM quota specified is too large to be provisioned into this cluster.\",\"replicaNumber\":\"Warning: you do not have enough data servers to support this number of replicas.\"},\"summaries\":{\"ramSummary\":{\"total\":3020947456,\"otherBuckets\":3019898880,\"nodesCount\":1,\"perNodeMegs\":1440,\"thisAlloc\":1509949440,\"thisUsed\":0,\"free\":-1508900864},\"hddSummary\":{\"total\":64408858624,\"otherData\":2563062770,\"otherBuckets\":13291574,\"thisUsed\":0,\"free\":61832504280}}}"], "warnings": []}

PLAY RECAP *********************************************************************
cb1                        : ok=6    changed=0    unreachable=0    failed=1
sg1                        : ok=15   changed=5    unreachable=0    failed=0
sg2                        : ok=15   changed=5    unreachable=0    failed=0

Traceback (most recent call last):
  File "libraries/provision/install_sync_gateway.py", line 259, in <module>
    install_sync_gateway(sync_gateway_install_config)
  File "libraries/provision/install_sync_gateway.py", line 167, in install_sync_gateway
    assert(status == 0)
AssertionError
```

Proposed fix: instead of trying to delete `data-bucket` and `index-bucket`, make a REST or command line API call to couchbase server to discover all of the buckets, and delete each one.

**Comment by sethrosetter** [comment]
@tleyden It does do that https://github.com/couchbaselabs/mobile-testkit/blob/master/libraries/testkit/server.py#L28-L62

**Comment by sethrosetter** [comment]
Is this failing during provisioning or cluster.reset()?

**Comment by tleyden** [comment]
Pretty sure it was cluster.reset()

**Comment by tleyden** [comment]
Sorry, take that back.  This is the command that triggers it:

```
python libraries/provision/install_sync_gateway.py --branch=feature/issue_1663_replicate && robot -b debug.txt  -v RESET_CLUSTER:False -v SERVER_VERSION:4.1.0 -v SYNC_GATEWAY_VERSION:1.2.0-79 -t "Test Stop Replication Via Replication Id" testsuites/syncgateway/functional/2sg_1cbs.robot
```

Full logs: https://gist.github.com/tleyden/6ce7dbe1d1708fc6f3fe94ae875b935c

The workaround is that I have to go manually delete the buckets in the Couchbase Server UI:

<img width="1118" alt="screen shot 2016-04-14 at 2 21 45 pm" src="https://cloud.githubusercontent.com/assets/296876/14544116/640117b6-024c-11e6-852b-5c143f7f1ab9.png">

and then re-run the above command.

**Comment by sethrosetter** [comment]
This should be resolved via - https://github.com/couchbaselabs/mobile-testkit/pull/657

**Comment by sethrosetter** [comment]
Closed as duplicate of #657
